### PR TITLE
chore: Organize files under `src/foundations/`

### PR DESCRIPTION
**Pull request by jaalah-akamai**
## Description 📝
Starting work for organizing our foundations to help build a cohesive and consistent visual language. This includes color palettes, typography, spacing, layout rules, and more

No real changes here, just moving things around really. There's a lot more to do and we'll fine tune things as we go, and as we have more discussions.

> Note: Same change was [previously reviewed](https://github.com/linode/manager/pull/9061) and merged into our feature branch, but we felt we could move these things over.

## Major Changes 🔄
- Created `foundations` folder where we will start organizing things better in separate files like: colors, fonts, darkTheme, lightTheme, breakpoints
- Created separate files for some base configs

### CODE REVIEW

**Review by abailly-akamai — APPROVED** [review]
Looks great - I really like this, looking forward to doing more work on the theme layer and design tokens!

File: packages/manager/src/foundations/breakpoints/index.ts
Comment (bnussman-akamai) [inline packages/manager/src/foundations/breakpoints/index.ts:1]: Should this just be `breakpoints.ts`? I don't think it's likely we will need a whole folder dedicated to breakpoints.

**Review by bnussman-akamai — APPROVED** [review]

File: docs/development-guide/01-repository-structure.md [hunk 1]
```diff
@@ -158,5 +158,5 @@ The /src directory has several subdirectories:
 
 1. Find where the styles are defined for the component you want to modify
    a. They are likely defined in the feature component's file, e.g. `src/features/<MyFeature>/<SomeComponent>.tsx`.
-2. Avoid making changes in `src/index.css`, `src/themeFactory.ts`, and `src/themes.ts` unless you are intentionally making a global styling change.
+2. Avoid making changes in `src/index.css`, `src/foundations/themes.ts`, and `src/themes/index.ts` unless you are intentionally making a global styling change.
```
Comment (dwiley-akamai) [inline docs/development-guide/01-repository-structure.md:161]: Will `src/themes/index.ts` exist?

Screenshot 2023-05-24 at 11 03 18 AM

File: docs/development-guide/13-coding-standards.md [hunk 2]
```diff
@@ -46,7 +46,7 @@ Function: handleLabelChange
 The styles for Cloud Manager are located in three places:
 
 - `packages/manager/src/index.css` contains global styles, utility styles, and accessibility related styles.
-- `packages/manager/src/themeFactory.ts` and `packages/manager/src/themes.ts` contain code for modifying the default [Material UI](https://mui.com) styles and theme specific styles.
-  - Light mode styles are located in `themeFactory.ts` and dark mode styles are located in `themes.ts`.
-  - The breakpoints can be modified at the end of `themeFactory.ts`.
-- Component-specific styles may be defined either at the end of the component file or in a dedicated file, named `ComponentName.styles.tsx`. Refer to the guidelines outlined in the "Styles" section of [Component Structure](02-component-structure.md#styles). 
\ No newline at end of file
+- `packages/manager/src/foundations/themes/themes.ts` and `packages/manager/src/themes/index.ts` contain code for modifying the default [Material UI](https://mui.com) styles and theme specific styles.
```
Comment (dwiley-akamai) [inline docs/development-guide/13-coding-standards.md:49]: similar question re: `src/themes/index.ts` as the previous comment

**Review by dwiley-akamai — APPROVED** [review]

File: packages/manager/src/utilities/theme.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { Theme } from '@mui/material/styles';
-import { ThemeName } from 'src/themeFactory';
-import { dark, light } from 'src/themes';
+import { ThemeName } from 'src/foundations/themes';
+import { dark, light } from 'src/foundations/themes';
```
[the remark this answers is not in the document]
Comment (jaalah-akamai) [reply packages/manager/src/utilities/theme.ts:3]: Yea that's what I meant!

**Pull request merged**